app/src/main/java/com/base/firebase_operations/MainActivity2.kt

```diff
@@ -0,0 +1,66 @@
+package com.base.firebase_operations
+
+import android.content.Intent
+import android.net.Uri
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.storage.ktx.storage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.tasks.await
+import kotlinx.coroutines.withContext
+
+class MainActivity2 : AppCompatActivity() {
+
+    private val Requestcode=0
+    private var currentfile: Uri?=null
+    lateinit var imageholder:ImageView
+    val imagereff=Firebase.storage.reference
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main2)
+        imageholder=findViewById(R.id.viewimage)
+        val upload: Button=findViewById(R.id.button5)
+        imageholder.setOnClickListener{
```

Comment:
Created a Imageview and set a onclick listener on it for getting the image from the gallery so just used the intent,
then set some extra information like here it would only display all the images present in the gallery.
Here we used also which is an scope function that allows us to perform additional operation on the object.

Reply:
Code block 1 reviewed